### ISSUE TITLE
Matterbridge Matrix Bridge not working since both updated

### ISSUE THREAD

**Issue by valvanet** [nextcloud/talk_matterbridge#61]
Hello Nextcloud-Community,

i got sort of the same Issue as described [here](https://github.com/nextcloud/spreed/issues/4282).
I believe it stopped working since I updated Nextcloud and my Matrix instancea few weeks ago.

As stated by OP of the thread linked above, the issue was gone after a while (no clou why)..
I guess my Issue could be a different rootcause, because it is not Telegram, but Matrix..

But I have no clou on how to troubleshoot this.

Matrix has had a recent change in Room versions (And a new Beta for 'Spaces'), but also with room version 6 (same version as from before the update) the bridge doesn't work anymore.. (I have not downgraded Matrix-Synapse tho)

I'm using the latest version of Matrix-Synapse (1.42.0)

Im using following versions of Talk/MB:
  Talk 12.1.1
  Talk Matterbridge 1.22.4 (

- Custom servers are configured (TURN/STUN)

> time="2021-09-16T18:13:27Z" level=info msg="Running version 1.22.3 e45bbe45" prefix=main
> time="2021-09-16T18:13:27Z" level=info msg="Parsing gateway myGateway" prefix=router
> time="2021-09-16T18:13:27Z" level=info msg="Starting bridge: matrix.0 " prefix=router
> time="2021-09-16T18:13:27Z" level=info msg="Connecting https://matrix.server.name" prefix=matrix
> time="2021-09-16T18:13:28Z" level=info msg="Connection succeeded" prefix=matrix
> time="2021-09-16T18:13:28Z" level=info msg="matrix.0: joining #channel-name:server.name (ID: #channel-name:server.namematrix.0)" prefix=matrix
> time="2021-09-16T18:13:28Z" level=info msg="Starting bridge: nctalk.1 " prefix=router
> time="2021-09-16T18:13:28Z" level=info msg=Connecting prefix=nctalk
> time="2021-09-16T18:13:28Z" level=info msg=Connected prefix=nctalk
> time="2021-09-16T18:13:28Z" level=info msg="nctalk.1: joining mc3wk2xs (ID: mc3wk2xsnctalk.1)" prefix=nctalk
> time="2021-09-16T18:13:28Z" level=fatal msg="Starting gateway failed: Bridge nctalk.1 failed to join channel: json: **cannot unmarshal array into Go struct field RoomInfo.ocs.data.lastMessage of type ocs.TalkRoomMessageData" prefix=main**

**Server configuration**

- Linux 4.19.0-17-amd64 Debian 4.19.194-3 (2021-07-18) x86_64
- Apache/2.4.48 (Debian) (apache2handler)
- pgsql PostgreSQL 13.2 (Debian 13.2-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
- PHP 8.0.10
- PHP Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, pdo_sqlite, session, posix, Reflection, standard, SimpleXML, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, apache2handler, apcu, bcmath, exif, gd, gmp, imagick, intl, ldap, memcached, pcntl, pdo_mysql, pdo_pgsql, redis, sodium, zip, Zend OPcache

**Nextcloud configuration:**

```
{
    "htaccess.RewriteBase": "\/",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "apps_paths": [
        {
            "path": "\/var\/www\/html\/apps",
            "url": "\/apps",
            "writable": false
        },
        {
            "path": "\/var\/www\/html\/custom_apps",
            "url": "\/custom_apps",
            "writable": true
        }
    ],
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "my-tld.de"
    ],
    "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
    "forwarded_for_headers": [
        "X-Forwarded-For",
        "HTTP_X_FORWARDED_FOR"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "pgsql",
    "version": "22.1.1.2",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "app_install_overwrite": [
        "occweb",
        "issuetemplate"
    ],
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpmode": "smtp",
    "mail_sendmailmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "465",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpsecure": "ssl",
    "overwriteprotocol": "https",
    "overwrite.cli.url": "https:\/\/my-tld.de",
    "loglevel": 2,
    "theme": "",
    "maintenance": false,
    "default_phone_region": "DE",
    "default_language": "de",
    "default_locale": "de_DE"
}
```

I found no related bug-report or anything in changelogs that could explain, why the "lastMessage" attribute is not set anymore..

Also I'm wondering if matrix or nextcloud is the cause..

Any help is much appreciated.

Have a nice evening
o/

**Comment by gary-kim** [comment]
Did you update recently? The presence of lastMessage should be optional now.

**Comment by gary-kim** [comment]
Yeah,you are on the most recent version. It should be fine if lastMessage is not set. Let me take a look at that.
